kotest-assertions/src/jvmTest/kotlin/com/sksamuel/kotest/properties/GenLinesTest.kt

```diff
@@ -0,0 +1,33 @@
+package com.sksamuel.kotest.properties
+
+import io.kotest.matchers.collections.shouldContainExactlyInAnyOrder
+import io.kotest.matchers.collections.shouldNotContainExactlyInAnyOrder
+import io.kotest.properties.Gen
+import io.kotest.properties.assertAll
+import io.kotest.properties.lines
+import io.kotest.specs.ShouldSpec
+
+class GenLinesTest : ShouldSpec() {
+    
+    
+    init {
+        should("Generate from lines input stream") {
+            val inputStream = List(4) { "line$it" }.joinToString(separator = "\n").byteInputStream()
+            
+            val lines = mutableSetOf<String>()
+            Gen.lines(inputStream).assertAll { line ->
+                lines += line
+            }
+            lines.shouldContainExactlyInAnyOrder("line0", "line1", "line2", "line3")
+        }
+    
+        should("Generate random different values and not sequenced") {
+            val inputStream = List(10000) { "$it" }.joinToString(separator = "\n").byteInputStream()
+            val lines = mutableListOf<String>()
+            Gen.lines(inputStream).assertAll(10000) { line ->
+                lines += line
+            }
```

Comment:
Is there a reason that this has to be `assertAll`, or could it just be `val lines = Gen.lines(inputStream).take(10000)`? Same question applies to the test above as well.